### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by diamond0910** [coltonstearns/dynamic-gaussian-marbles#1]
Thank you for your great work!

I meet some problem during training the model.

I follow this repo and install cuda 11.7. This is the log of 'nvcc -V'.
![image](https://github.com/user-attachments/assets/0daac25f-b7fc-440c-8496-e7e3246ed914)

And the cuda driver verision is 545.23.06.
![image](https://github.com/user-attachments/assets/776682a6-c3a7-4e1a-adb5-617e295ab942)

When I run the command 'CUDA_VISIBLE_DEVICES=1 python train.py --data_dir ./data/paper-windmill --model_config configs.dgmarbles_dycheck_wpose'. I get 'Segmentation fault (core dumped)' error.

![image](https://github.com/user-attachments/assets/0c14444e-ece3-4769-b4dd-fc861458497c)

Can you give me some advice? Thank you !

**Comment by zhouwy19** [comment]
To verify whether the problem is with the cuda version, I trained a simple example of mnist and found it to be normal.
![image](https://github.com/user-attachments/assets/3bfd5134-d397-4da6-9884-12f55960c8ab)

And I find the Segmentation fault (core dumped) error occurs in the execution of this sentence "mod = importlib.import_module(args.model_config)" (src.utils.load_model.py#27)

**Comment by coltonstearns** [comment]
Hi, thank you for raising this issue! I tried to reproduce it on my side, but was unable to... I used a V100 with driver version  525.105.17 and with CUDA 11.7. Could you try a few other things to try that may be causing problems:

- Check that `which nvcc` points to `/usr/local/cuda/bin/nvcc`. Or if not, ensure that `/usr/local/cuda` is also CUDA 11.7. (Sometimes anaconda overwrites the default nvcc command).
- What gcc version are you using? (`gcc --version`). I've tested the code on 9.3.0 and 11.4.0.

Unfortunately, "mod = importlib.import_module(args.model_config)" is where we dynamically load the code, so the seg fault could be anywhere. However, my intuition is that it would stem from one of the  `frnn`, `pytorch3d`, or `simple_knn` packages.

To check could you:
- Comment out line 13 in `src/models/gaussians.py` and line 15 in `src/utils/tracking_utils.py`, and then run  `CUDA_VISIBLE_DEVICES=1 python train.py --data_dir ./data/paper-windmill --model_config configs.dgmarbles_dycheck_wpose --isometry_loss_weight 0.0 --tracking_loss_weight 0.0`. If this runs, the issue is with the FRNN package.
- Commeng out lines 11-12 in `src/models/gaussians.py` and line 34 in `src/models/model.py`, and then run `CUDA_VISIBLE_DEVICES=1 python train.py --data_dir ./data/paper-windmill --model_config configs.dgmarbles_dycheck_wpose`. This will eventually error, but if it successfully starts running, it will imply the error segfault is coming from PyTorch3D.

Let me know if any of this works!

**Comment by diamond0910** [comment]
Thank you for your reply.

I successfully ran this code on CUDA 12.3 and found that FRNN can be successfully installed on 12.3. So I think you can modify the installation tutorial. But I'm not sure if there's a decrease in effectiveness.

![image](https://github.com/user-attachments/assets/772c4822-1a18-4529-88aa-2e202bbbb68c)

At the same time, I found that training on the paper-windmill dataset seemed to take six hours, a little long, is that normal?

![image](https://github.com/user-attachments/assets/a9e94759-def6-427e-919f-578b8048a2ea)

Thank you again.

**Comment by coltonstearns** [comment]
Ok great to know it can work with CUDA 12.3 - I'll update this. Initially I ran into compilation errors when I tried this with CUDA 12.X

As for runtime, that actually seems about right for a V100 GPU. Using newer GPUs has a pretty large impact on runtime. For instance, on an RTX 3090 right now, I'm getting ~3.5 hours.

![image](https://github.com/user-attachments/assets/f5a17220-47bf-44c1-8883-0cd16c92c931)

Also, the DyCheck dataset with camera pose is by far the slowest configuration to train, because (1) the videos are pretty long in the ~270-500 frame range, and (2) this configuration sets the divide-and-conquer to learn background scene dynamics that span the entire sequence (so we cannot stop training earlier at shorter subsequences).

Let me know if you have any other questions about this or concerns! Otherwise, I'll leave it to you to close the issue.
